lib/Jsrt/Jsrt.cpp

```diff
@@ -4290,36 +4307,30 @@ _ALWAYSINLINE JsErrorCode CompileRun(
         }
     }
 
-    LoadScriptFlag scriptFlag;
-    const byte* script = isExternalArray ?
-        ((Js::ExternalArrayBuffer*)(scriptVal))->GetBuffer() :
-        (const byte*)((Js::JavascriptString*)(scriptVal))->GetSz();
-    const size_t cb = isExternalArray ?
-        ((Js::ExternalArrayBuffer*)(scriptVal))->GetByteLength() :
-        ((Js::JavascriptString*)(scriptVal))->GetLength();
+    JsErrorCode error = GlobalAPIWrapper_NoRecord([&]() -> JsErrorCode {
+        if (isString)
+        {
+            Js::JavascriptString* jsString = Js::JavascriptString::FromVar(scriptVal);
+            script = (const byte*)jsString->GetSz();
 
-    if (isExternalArray && isUtf8)
-    {
-        scriptFlag = (LoadScriptFlag) (LoadScriptFlag_ExternalArrayBuffer | LoadScriptFlag_Utf8Source);
-    }
-    else if (isUtf8)
-    {
-        scriptFlag = (LoadScriptFlag) (LoadScriptFlag_Utf8Source);
-    }
-    else
-    {
-        scriptFlag = LoadScriptFlag_None;
-    }
+            // JavascriptString is 2 bytes (wchar_t/char16)
+            cb = jsString->GetLength() * sizeof(wchar_t);
```

Comment:
>cb = jsString->GetLength() * sizeof(wchar_t); [](start = 12, length = 45)

This doesn't work if the scriptVal was created using JsCreateStringUtf16. That means if we pass a scriptVal created using JsCreateStringUtf16 we should pass JsParseScriptAttributeArrayBufferIsUtf16Encoded and then not do this math but that doesn't look right.